controllers/front/ProductController.php

```diff
@@ -1091,16 +1091,27 @@ function ($elem) use ($attribute) {
      */
     private function getIdProductAttributeByGroup()
     {
-        $groups = Tools::getValue('group');
-        if (empty($groups)) {
-            return null;
+        try {
```

Comment:
I remember we discussed about this one and we realized that the whole attribute fetching feature is meaningless for product which have no combinations. Fixing this method is a good thing, but it's a side effect of the initial problem.

In `getIdProductAttributeByGroupOrRequestOrDefault` we should check that product has combinations in the first place, if it doesn't the method should return 0 early.

Reply:
Agreed I forgot that part sorry. I am just nore sure about returning early 0 in `getIdProductAttributeByGroupOrRequestOrDefault ` i'd rather return `null` seing the return result usage a line 546 and 562